### ISSUE TITLE
Value-based marketing in the bounty system

### ISSUE THREAD

**Issue by casparlusink** [rchain/bounties#926]
As discussed in #695 and #775, by @philipandri002, @pmoorman, @kcdc, @David405, @deannald and others (please add), the bounty program could benefit from a more value-based approach to marketing. I created a new issue to address this in detail.

<h2>Summary</h2>
By attributing a value to a new member of the RChain coop, marketing efforts can be measured effectively and the community can be easily rewarded for bringing in new developers, RAM's, dApp startups, etc. Below I elaborate on how to set up such a process and address the concerns that have been raised before.

<h2>Benefit to RChain</h2>
By creating a measurable attribution of 'value' for new members, the scalability and efficiency of RChain's community marketing channels increases, the bounty system becomes more fair, transparent and less prone to abuse.
Incentivizing community members to bring in new valuable people, can increase member growth (virality) of RChain as a whole.

<h3>Concerns</h3>

1. As brought up by @jasoncruzzy  (read here: <a href="https://github.com/rchain/bounties/issues/775#issuecomment-398361016">issue 775</a>)
    1. The valuation of RAM acquisition is prone to abuse and an influx of NRAM’s 
    2. Proposition to implement a marketing category based approach to this issue

2. As brought up by @David405 (read here: <a href="https://github.com/rchain/bounties/issues/695#issuecomment-390477539">issue 695</a> )
    1. The ability to scam the bounty system with an affiliate link of some sort
    2. The possibility of bad marketing efforts like spamming through affiliate email lists etc.

3. The general privacy concern within the community of using tools to track members-activity.

These are important issues and they should be addressed (and solved) before we can start. In the following proposal I try to address these issues as clearly as possible. Please comment if you find more concerns.

<h2>Proposal</h2>
For the complete project I identified 3 subgoals: 

- **Subgoal 1:** Establish a value estimation table for each type of valuable members. As discussed in #775, a (rough) estimate for all the different types of members is needed. 
- **Subgoal 2:** Create an measurable process for community members to refer new customers. As in issue #695: an affiliate link to measure the acquisition efforts of community members.
- **Subgoal 3:** Design a process to validate newly acquired members, which is cost-effective and scalable. To combine subgoal 1 and 2 into a smooth system, a process is needed to cross-check referred customers without taking up to many resources from the coop.

<h3>Subgoal 1</h3>
We can take the table of @pmoorman (this one: <a href="https://github.com/rchain/bounties/issues/775#issuecomment-398317818">Value Table</a>) for now, with some iterations that can be discussed further.
Also note that it probably changes over time, since our knowledge and the state of things evolves.

*To address concern 1.1*, it might be smart to set the value of a single RAM to $20, to make any scam attempts useless. This way, bringing in one (fake) RAM will cost as much as it pays.

<h3>Subgoal 2</h3>
There’s currently no way to measure marketing channels or lead behavior, or to consistently check who referred who. As proposed by @philipandri002, the use of a referral link might be the way to go. 

To effectively implement such a process, we need a tool to track new leads through the acquisition and onboarding. Here you can find the technical plan to set up such a measuring process:
<a href="https://docs.google.com/document/d/1W9YPVuve8Uzdw_P1G6wlRTTW6RNoouBH80ovhd-kqbM/edit?usp=sharing">Technical Plan</a>.

With such a process better marketing analytics can be acquired and value can be more easily assigned to channels/individual members.

*To address the privacy concern (3)*  of tracking new members on our website, these tools offer ways to anonymously track leads throughout the process with a so-called 'alias'. As soon as they sign-up, the alias can be assigned to the member’s profile in the data-base. This way the identity remains anonymous for marketing purposes, but can be used by KYC validators to check the referral identity. More info in technical plan^.

*To address concern 1.2*, using a marketing channel based incentivization scheme might not cover the variability of different channels. Different marketing channels have different purposes (as in, bring in different kinds of people) and there is a high variability in the quality and quantity of leads the channels bring in individually. Therefore, such a process is more open to abuse. In short, incentivizing per channel versus per lead, impedes the possibility to measure the effectiveness of different channels. A value-based referral scheme solves these problems.

<h3>Subgoal 3</h3>
To make this whole process workable and secure, the new leads need to be validated for their value, cost-effectively in a scalable process. Clearly, validating new leads will consume resources, but to minimize those efforts, the process must be self-driven. Specifically, the members that bring in new leads must report those leads (through their referral link metrics or other technical integration), providing evidence for their activity (as in if they are an active member or not), their role (RAM/dev/etc), competence and past working experience. <br><br>

_Cost-effective_
We could use a cascade incentivization method, where every new lead is payed the minimum ($20 for RAM) and the additional amount for every piece of evidence presented. This evidence needs to be presented in a systematic manner, like with a checklist of evidence that will be emailed to the appropriate validator, or through an expenses form (on google docs i.e.). This way, the validators job is minimized, making the process cost-effective.

_Scalable_
To address the scalability factor, the validators job must not only be minimal, but also less thorough. If we impose a punishment (like being kicked as a member, or financial punishment by way of deposit) for abuse of the system, in later stages, the process can become more automatized or use a more 'random-sample' way of validating.

*To address concern 2.1*, if proof is required to be provided by the members that bring in the leads, the possibility of scamming is minimized, since it takes considerably more effort. The $20 reward for RAM’s also makes scamming idle.

*To address concern 2.2*, also here the implementation of punishments for misbehavior can counter spamming efforts. Additional measures can be taken, by asking referrals to report the place where they found the link. This way if bad marketing practices are used, this is noticed quickly.

<h2>Conclusion</h2>
Setting in place a process to make referral efforts for community members a success, could be an effective way to enhance coop’s marketing and creates a measurable and scalable marketing process. If this incentivization scheme succeeds, other process/marketing channels can possibly make use of schemes.

---

I don’t know what this project will look like budget-wise, but I think it should be carried out better sooner than later. 

@pmoorman already mentioned: to move forward with this issue the proposal needs to be agreed on and approved (in a meeting of some sort) so that we can start working on it.

<h2>Action plan</h2>

* Value table evaluation and finalization (sub 1): @pmoorman?

* Technical plan roll out (sub 2): I could work on this

* Setup validation process (sub 3): @ian @KYCvalidators, I could also help here

    * Documentation (checklists etc.) on how to prove your leads

    * Email account for prove

    * Managerial process for validation -> payment

    * Integration with the marketing tool

* Assign validators @pmoorman

* Inform community/new members of process @anyone that wants to help, I could also help here

**Comment by owans** [comment]
Very elaborate and beneficial for the co-op in the long term.

**Comment by philipandri002** [comment]
@casperlusink, I agree this is a more detailed expressions. @dckc maybe you should take a look now and see were you can be of help.

**Comment by pmoorman** [comment]
@casparlusink @philipandri002 I have put this on the agenda for the TAC (= label guides committee, #925) which will meet later today.

**Comment by pmoorman** [comment]
Update regarding the TAC call: this topic wasn't addressed on the call, but I discussed it briefly with Dan afterwards. 

Contrary to my understanding, there isn't any policy in the coop that would impair on the suggested approach (I have updated my [#219 comment](https://github.com/rchain/bounties/issues/219#issuecomment-399946056) to reflect the new insight). Given that situation, this topic is just a marketing-only problem, and something we should be able to solve within marketing.

@casparlusink @philipandri002 
**My suggestion would be that you go ahead and contact the relevant people that can help us implement tracking as needed.**

Those people are: 

- Kevin & Craig from the marketing org. Kevin is responsible for the website work, and Craig is doing development on it. There is a public repository [here](https://github.com/rchain/marketing), where the website code lives. I can also put you in touch with Kevin (send me a DM).

- We would need to figure out who's responsible for the KYC process (Ian, I think), and how we could best tap into that. For that to be successful, provide an precise description of what's exactly needed from them, keeping privacy concerns in mind.

- Finally, it's probably a good idea to include Ankur & Patrick on this topic, so they're aware of what's going on. I think @kitblake would be the best liaison, since he's a bounty system marketing guide, too.

- There might be other areas to consider, but that's your call. This is just the first stuff that comes to mind.
